src/cpp/src/text2image/stable_diffusion_3_pipeline.hpp

```diff
@@ -0,0 +1,576 @@
+            }
+
+            auto scheduler_step_result = m_scheduler->step(noisy_residual_tensor, latent, inference_step);
+            latent = scheduler_step_result["latent"];
+        }
+
+        float* latent_data = latent.data<float>();
+        for (size_t i = 0; i < latent.get_size(); ++i) {
+            latent_data[i] = (latent_data[i] / m_vae_decoder->get_config().scaling_factor) +
+                             m_vae_decoder->get_config().shift_factor;
+        }
+
+        return m_vae_decoder->infer(latent);
+    }
+
+private:
+    bool do_classifier_free_guidance(float guidance_scale) const {
+        return guidance_scale > 1.0;
+    }
+
+    void initialize_generation_config(const std::string& class_name) override {
+        assert(m_transformer != nullptr);
+        const auto& transformer_config = m_transformer->get_config();
+
+        size_t vae_out_channels_size = m_vae_decoder->get_config().block_out_channels.size();
+        const size_t vae_scale_factor = std::pow(2, vae_out_channels_size - 1);
+        m_transformer->set_vae_scale_factor(vae_scale_factor);
+
+        m_generation_config.height = transformer_config.sample_size * vae_scale_factor;
+        m_generation_config.width = transformer_config.sample_size * vae_scale_factor;
+
+        if (class_name == "StableDiffusion3Pipeline") {
+            m_generation_config.guidance_scale = 7.5f;
+            m_generation_config.num_inference_steps = 50;
+        } else {
+            OPENVINO_THROW("Unsupported class_name '", class_name, "'. Please, contact OpenVINO GenAI developers");
+        }
+    }
+
+    void check_image_size(const int height, const int width) const override {
+        assert(m_transformer != nullptr);
+        const size_t vae_scale_factor = m_transformer->get_vae_scale_factor();
+        OPENVINO_ASSERT((height % vae_scale_factor == 0 || height < 0) && (width % vae_scale_factor == 0 || width < 0),
+                        "Both 'width' and 'height' must be divisible by",
+                        vae_scale_factor);
+    }
+
+    void check_inputs(const GenerationConfig& generation_config) const override {
+        check_image_size(generation_config.width, generation_config.height);
+
+        const bool is_classifier_free_guidance = do_classifier_free_guidance(generation_config.guidance_scale);
+        const char* const pipeline_name = "Stable Diffusion 3";
+
+        OPENVINO_ASSERT(is_classifier_free_guidance || generation_config.negative_prompt.empty(),
+                        "Negative prompt is not used when guidance scale <= 1.0");
+        OPENVINO_ASSERT(is_classifier_free_guidance || generation_config.negative_prompt_2.empty(),
+                        "Negative prompt 2 is not used when guidance scale <= 1.0");
+        OPENVINO_ASSERT(is_classifier_free_guidance || generation_config.negative_prompt_3.empty(),
+                        "Negative prompt 3 is not used when guidance scale <= 1.0");
```

Comment:
as we have not enabled T5Encoder yet, we probably need to assert that prompt_3 and negative_prompt_3 are not passed ?

Reply:
let's add this check as well https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py#L532-L533

Reply:
Updated `check_image_size(...)`